libraries/chain/apply_context.cpp

```diff
@@ -1229,7 +1229,14 @@ void apply_context::add_ram_usage( account_name account, int64_t ram_delta, cons
    }
 }
 
-void apply_context::add_disk_usage( account_name account, int64_t disk_delta, const storage_usage_trace& trace ) {
+void apply_context::update_disk_usage( account_name account, int64_t disk_delta, const storage_usage_trace& trace ) {
+
+   if (disk_delta > 0 && !privileged) {
+      EOS_ASSERT(receiver == act->account, subjective_block_production_exception,
```

Comment:
This should not be a subjective failure.  The similar check in `update_db_usage` is a subjective mitigation for before `RAM_RESTRICTIONS` is activated.  Since DISK in new, we don't need subjective mitigations and can always apply the restriction.  The actual check that you should copy is in `exec_one`.

Reply:
The behaviour of the check in `exec_one` is slightly different from the subjective check in `update_db_usage`.  Unless we've decided that that was a mistake it would be better to change `exec_one`.